### DISCUSSION TITLE
Business rule running only for primary property changes

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3391] [Questions] [answered]
I have a business rule that inherits from CommonBusinessRule.
It accepts three input properties. During testing, I noticed that my business rule runs when the object is created but afterward, it only re-runs for primary property changes.
Just to make sure, I put a breakpoint on SetProperty for the other two input properties and UI changes are indeed calling them.
Do I need to create a Dependency rule?

Kind Regards

**Comment by rockfordlhotka** [comment, marked as answer]
Yes, you need dependency rules. A rule only runs based on the primary property.

**Reply by Chicagoan2016** [reply]
Thank you @rockfordlhotka

**Reply by Chicagoan2016** [reply]
@rockfordlhotka I tried using the 'DependencyFrom' (@jonnybee ) rule and 'Dependency' from Csla.Rules.Commonrules but so far no luck. 

In my case, the three input properties to the rule take part only in this one rule.
I want the rule to run for changes in any of the input properties, not just primary property.
Since the other input properties don't have any other business rule associated. I can't seem to work the Dependency route.

Kind regards.

**Reply by rockfordlhotka** [reply]
I am guessing you got it working since this is marked as answered?
